lib/kitchen/driver/gce.rb

```diff
@@ -373,13 +374,15 @@ def machine_type_url
         "zones/#{zone}/machineTypes/#{config[:machine_type]}"
       end
 
-      def instance_metadata
-        metadata = {
-          "created-by"            => "test-kitchen",
-          "test-kitchen-instance" => instance.name,
-          "test-kitchen-user"     => env_user,
-        }
+      def metadata
+        config[:metadata].merge({
+                                    "created-by"            => "test-kitchen",
+                                    "test-kitchen-instance" => instance.name,
+                                    "test-kitchen-user"     => env_user,
+                                })
+      end
 
+      def instance_metadata
         Google::Apis::ComputeV1::Metadata.new.tap do |metadata_obj|
           metadata_obj.items = metadata.each_with_object([]) do |(k, v), memo|
             memo << Google::Apis::ComputeV1::Metadata::Item.new.tap do |item|
```

Comment:
With this addition to allow a hash of metadata values, what do you think about explicitly calling `.to_s` on the `k` and `v` objects below in this loop to ensure that the `Google::Apis::ComputeV1::Metadata::Item` instances created are strings. Just in case someone gets excited and tries to put nested hashes in their metadata.

Reply:
Sounds like a good idea. Adding ...